### DISCUSSION TITLE
How to forbid keys auto-sort (in m:auto) in JSON request/response view?

### DISCUSSION THREAD

**Discussion by pozernishku** [mitmproxy/mitmproxy#5853] [Q&A] [answered]
Here is what I mean.

This screenshot shows the `m:raw` JSON (note the order) view.
<img width="1448" alt="Снимок экрана 2023-01-07 в 16 16 05" src="https://user-images.githubusercontent.com/14317371/211155355-82d02bc3-5143-4fa5-a0cf-4a57493062c9.png">

But here how it looks in `m:auto` JSON view (the same thing is in `Response` tab which is more important for me).
<img width="1450" alt="Снимок экрана 2023-01-07 в 16 15 47" src="https://user-images.githubusercontent.com/14317371/211155450-11f18cba-edcd-487a-a775-2d84429aacc5.png">

Is it possible to keep this colourful JSON view without sorting keys?

**Comment by mhils** [comment, marked as answer]
We may just need to remove `sort_keys` here: https://github.com/mitmproxy/mitmproxy/blob/d06afcc1f98a04065c656197768c2b6f3ef40d0d/mitmproxy/contentviews/json.py#L22

Not sure why it's there - reusing the existing order sounds most sensible to me. If the approach above works, please file a PR! 😃

**Comment by pozernishku** [comment]
@mhils thank you, but I'm not sure about PR right now.

**Comment by aoyama-val** [comment]
> reusing the existing order sounds most sensible to me

+1
